autonomy/deploy/generators/localhost/base.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2024 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Localhost Deployment Generator."""
+import json
+import shutil
+import subprocess  # nosec
+import typing as t
+from pathlib import Path
+
+from aea.configurations.constants import (
+    DEFAULT_LEDGER,
+    LEDGER,
+    PRIVATE_KEY,
+    PRIVATE_KEY_PATH_SCHEMA,
+)
+from aea.helpers.io import open_file
+from aea.helpers.yaml_utils import yaml_load_all
+
+from autonomy.deploy.base import BaseDeploymentGenerator
+from autonomy.deploy.constants import DEFAULT_ENCODING
+from autonomy.deploy.generators.localhost.utils import check_tendermint_version
+
+
+class HostDeploymentGenerator(BaseDeploymentGenerator):
+    """Localhost deployment."""
+
+    output_name: str = "agent.json"
+    deployment_type: str = "localhost"
+
+    def generate_config_tendermint(self) -> "HostDeploymentGenerator":
+        """Generate tendermint configuration."""
+        tmhome = str(self.build_dir / "node")
+        tendermint_executable = check_tendermint_version()
+        # if executable found, setup its configs
+        subprocess.run(  # pylint: disable=subprocess-run-check # nosec
+            args=[
+                tendermint_executable,
+                "--home",
+                tmhome,
+                "init",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+
+        # TODO: Dynamic port allocation
```

Comment:
Let's open an issue for this TODO if not addressed.

Reply:
created one here #2259